front/lib/resources/data_source_resource.ts

```diff
@@ -308,7 +308,7 @@ export class DataSourceResource extends ResourceWithVault<DataSource> {
     await AgentTablesQueryConfigurationTable.destroy({
       where: {
         dataSourceWorkspaceId: auth.getNonNullableWorkspace().sId,
-        dataSourceId: this.id,
+        dataSourceId: this.id.toString(),
```

Comment:
Actually this does not look to be right. On this table `dataSourceId` is the name of the data source. Will send a PR.